product docs toolset to blog (#9061)
fishtown-analytics/docs.getdbt.com: website/blog/2026-04-29-devil-in-the-docs.md

---
title: "The devil is in the docs"
description: "The devil in the details and the details are the docs that we write."
slug: the-devil-in-the-docs
authors: [mirna_wong]
tags: [ai, docs]
hide_table_of_contents: false
date: 2026-04-30
is_featured: true
---

> *"By all means, move at a glacier's pace."*
> — Miranda Priestly, *The Devil Wears Prada*

> There's another scene in *The Devil Wears Prada* that I think about more than that one. If you haven't seen the it yet, I'll try not to spoil it for you. Miranda (the character modeled after Anna Wintour) turns to Andy (her assistant) and explains, with _complete_ patience, that the [cerulean blue](https://en.wikipedia.org/wiki/Cerulean) in Andy's "lumpy sweater" didn't come from nowhere &mdash; it's traced back through fashion decisions made years earlier, by people who thought carefully about every choice. The whole chain, invisible to Andy, was built on deliberate _curation_.

Reviewer: Is it Andy and not Andi? Just checking

Author: it's andy, short for andrea! good question, i can elaborate https://devil-wears-prada.fandom.com/wiki/Andrea_Sachs

> There's another scene in *The Devil Wears Prada* that I think about more than that one. If you haven't seen the it yet, I'll try not to spoil it for you. Miranda (the character modeled after Anna Wintour) turns to Andy (her assistant) and explains, with _complete_ patience, that the [cerulean blue](https://en.wikipedia.org/wiki/Cerulean) in Andy's "lumpy sweater" didn't come from nowhere &mdash; it's traced back through fashion decisions made years earlier, by people who thought carefully about every choice. The whole chain, invisible to Andy, was built on deliberate _curation_.

Reviewer: Maybe something shorter like, "There's another scene I find more compelling."

> "didn't come from nowhere"
Are you using double negative because that's how it's written in the movie? If not, I suggest something more concrete, such as didn't come out of thin air or wasn't random or something like that.

> There's another scene in *The Devil Wears Prada* that I think about more than that one. If you haven't seen the it yet, I'll try not to spoil it for you. Miranda (the character modeled after Anna Wintour) turns to Andy (her assistant) and explains, with _complete_ patience, that the [cerulean blue](https://en.wikipedia.org/wiki/Cerulean) in Andy's "lumpy sweater" didn't come from nowhere &mdash; it's traced back through fashion decisions made years earlier, by people who thought carefully about every choice. The whole chain, invisible to Andy, was built on deliberate _curation_.

Reviewer: ```suggestion
There's another scene in *The Devil Wears Prada* that I find more compelling. If you haven't watched the movie, I won't spoil it for you. Miranda (the character modeled after Anna Wintour) turns to Andy (her assistant) and explains, with _complete_ patience, that the [cerulean blue](https://en.wikipedia.org/wiki/Cerulean) in Andy's "lumpy sweater" didn't come from nowhere &mdash; it can be traced back through fashion decisions made years earlier, by people who thought carefully about every choice, right down to the exact blue they would make the sweater in. The whole chain, invisible to Andy, was built on deliberate _curation_.
```

> That iconic scene reminds me of what documentation architecture feels like to me. Users (could be developers, analysts, data engineers, etc.) ask a question, get an answer, keep working. They don't need to see the decisions behind it &mdash; what to include, how to structure it, where the gaps are, what needs updating. But those decisions shape every answer and experience they get, especially the ones coming from AI.

Reviewer: The first sentence could use some tightening.
```suggestion
That iconic scene is how I'd describe documentation architecture. Users (perhaps a developer, analyst, data engineer) have a question, look at the docs, get an answer, keep working. They don't need to see the decisions behind it &mdash; what to include, how to structure it, where the gaps are, what needs updating. But those decisions shape every answer and experience they get, especially the ones coming from AI.
```

> This blog discusses how we as a docs team are trying to bring docs closer to users; and why that architecture matters more than ever in the AI era.

Reviewer: I suggest making this "docs teams" meaning all docs teams, but other places where you're referrign to us I'd change to "Docs team"
>  "According to the Microsoft Style Guide, it would be Docs team — lowercase "team."
```suggestion
This blog discusses how docs teams bring documentation closer to users; and why that architecture matters more than ever in the AI era.
```

Author: hey @runleonarun i can absolutely see your point here -- the intent of the blog was more to write about the journey our docs team took in getting the docs closer to users, which resulted in the dbt mcp server. at first, i feel hesitant making it about 'all docs teams wanting to bring docs closer to users' because this blog is particular to our journey, which might not be applicable to other teams. but maybe it's worth generalizing?

Author: actually i re-read and changed it! good shout!

Think of Andy in *The Devil Wears Prada* &mdash; constantly fielding requests for information she could theoretically find, but scrambling each time because it isn't at her fingertips. She's not uninformed. She's just not connected to the right source when it counts.

That was the [dbt MCP server's](https://docs.getdbt.com/docs/dbt-ai/about-mcp) relationship with the docs:

- The open source documentation at [docs.getdbt.com](https://docs.getdbt.com) is carefully maintained (jointly by the docs team _and_ our amazing dbt community), up-to-date, and formatted for humans, of course. And now machine consumption: there's an `llms.txt` index, a full-content flat file, and markdown output on every page. We have an `AGENTS.md` file that lists how to access the docs via web requests. The source was solid and accessible.
- The dbt MCP server &mdash; where dbt users interact with dbt through AI tools &mdash; couldn't reach any of it by default. It has eight toolset categories: CLI, Semantic Layer, Discovery, Admin API, SQL, Codegen, Fusion, Server Metadata. But none of them connected to the live docs by default. It didn't have our docs as its canonical source.

> So when an agent was asked *"how do I configure incremental models?"*, it improvised or used training data, pattern matching, best guess. The docs kinda existed. The path to them didn't.

Reviewer: ```suggestion
So when an agent was asked *"how do I configure incremental models?"*, it improvised or used training data, pattern matching, best guesses. The docs kinda existed. The path to them didn't.
```

> So when an agent was asked *"how do I configure incremental models?"*, it improvised or used training data, pattern matching, best guess. The docs kinda existed. The path to them didn't.

Reviewer: Can you flesh out or reframe "The docs kinda existed." It feels a little vague. They kinda existed to the AI?

Author: The docs existed. A direct path to them, to the canonical source, didn't.

> <Lightbox src="/img/blog/2026-04-30-devil-in-the-docs/dwp-meme.png" title="Miranda mad at AI for hallucinating about incremental models" />

Author: ```suggestion
<Lightbox src="/img/blog/2026-04-30-the-devil-in-the-docs/dwp-meme.png" title="Miranda mad at AI for hallucinating about incremental models" />
```

> <Lightbox src="/img/blog/2026-04-30-the-devil-in-the-docs/dwp-meme.png" title="Miranda mad at AI for hallucinating about incremental models" />

Author: ```suggestion
<Lightbox src="/img/blog/2026-04-30-the-devil-is-in-the-docs/dwp-meme.png" title="Miranda mad at AI for hallucinating about incremental models" />
```

## What this changes day to day
- Before: write a macro, hit an unfamiliar function, alt-tab to a browser, search docs, read the page, return to your editor, try to remember where you were. Or worse, ask AI and get a response that is not grounded in actual current documentation.(hallucinate much?)
- After: ask Claude or your AI tool directly, get an answer grounded in actual current documentation, keep coding!

We also added the product docs toolset to dbt's [Developer agent](https://docs.getdbt.com/docs/dbt-ai/developer-agent?version=2.0) experience &mdash; bringing the docs closer to users in dbt platform and the Studio IDE. 🎉

The Book arrived. No context switch required.

For analysts exploring a shared project, it means understanding what a model does without navigating to a separate tab. For teams working across different dbt setups, it means consistent, authoritative answers regardless of where they're working or who's asking. For the docs team, it means the work we put into writing and maintaining docs.getdbt.com is doing more than it was before &mdash; reaching users where they actually are.

## What's next

> This is the first page of any in this new chapter. But there are a few things on the radar for the next pages as we continue to try to improve the docs experience:

Author: ```suggestion
This is the first page of many in this new chapter. But there are a few things on the radar for the next pages as we continue to try to improve the docs experience:
```

> This is the first page of many in this new chapter. But there are a few things on the radar for the next pages as we continue to try to improve the docs experience:

Reviewer: ```suggestion
This is the first page of many in this new chapter. But we're planning a few things for the next pages as we continue to try to improve the docs experience:
```

- **Version-aware docs fetching** &mdash; Right now these tools return current docs. A developer on dbt Core 1.10 asking about incremental strategies gets 2026 docs. Version-aware routing &mdash; returning the right page for the right dbt version &mdash; is the next meaningful improvement, and we're [working](https://github.com/dbt-labs/dbt-mcp/pull/638) through it now!
- **Smarter search ranking** &mdash; Relevance is good. Relevance tuned to dbt-specific concepts and query patterns would be better.
- **Coverage gaps as a signal** &mdash; MCP usage will surface which questions return weak results &mdash; pages that are missing, thin, or outdated. That's a direct queue for the docs team, and a feedback loop that didn't exist before.
- **A dedicated dbt docs MCP server** maybe? &mdash; It's not the right call right now &mdash; one coherent product with an active existing server is the right place to start &mdash; but in the future, maybe a standalone docs server could reach every MCP client regardless of whether they use the dbt MCP server at all?

## Give it a try!

> - If you're new to the MCP server, the [quickstart guide](https://docs.getdbt.com/docs/dbt-ai/mcp-quickstart-oauth?version=2.0) walks through both local and remote configuration.

Reviewer: should we change the links to be relative?

Author: you know all this time i thought relative links didn't work in the blog and i just tested it and learned it does!

We'd love to hear your feedback! Open an issue at [github.com/dbt-labs/docs.getdbt.com](https://github.com/dbt-labs/docs.getdbt.com) to flag any bugs, typos, wrong info, etc. &mdash; or find me in the [dbt community Slack](https://www.getdbt.com/community/join-the-community). 

## Conclusion

Miranda's point about the cerulean sweater wasn't really about fashion. It was about invisible chains &mdash; every deliberate decision ripples further than the person making it can see, and that the quality of the source shapes everything downstream.

That's where docs are now. A developer asks an AI a question and gets an answer without possibly ever opening a browser. They don't need to see the decisions behind it &mdash; the careful decisions, curation, structure, accuracy, the gaps that got filled. But those decisions are in the chain. They always were. The difference is that now, they travel faster and reach further than ever before.

We're no longer moving at a glacial pace. The current runs _fast_. And the devil? Well, it's in the docs.

> *Mirna Wong is a technical writer at dbt Labs who loves em-dashes, pop culture references, and writing about dbt and AI. dbt's product docs are open source &mdash; contributions and issues are **always** welcome at [github.com/dbt-labs/docs.getdbt.com](https://github.com/dbt-labs/docs.getdbt.com).*

Author: ```suggestion
*Mirna Wong is a technical writer at dbt Labs who loves em-dashes, pop culture references, and writing about dbt and AI. dbt's product docs are open source &mdash; contributions and issues are **always** welcome at [github.com/dbt-labs/docs.getdbt.com](https://github.com/dbt-labs/docs.getdbt.com/issues).*
```

## What this changes day to day
- Before: write a macro, hit an unfamiliar function, alt-tab to a browser, search docs, read the page, return to your editor, try to remember where you were. Or worse, ask AI and get a response that is not grounded in actual current documentation (hallucinate much?).
- After: ask Claude or your AI tool directly, get an answer grounded in actual current documentation, keep coding! Docs 🤝 canonical source.

We also added the product docs toolset to dbt's [Developer agent](/docs/dbt-ai/developer-agent?version=2.0) experience &mdash; bringing the docs closer to users in dbt platform and the Studio IDE. 🎉

The Book arrived. No context switch required.

For analysts exploring a shared project, it means understanding what a model does without navigating to a separate tab. For teams working across different dbt setups, it means consistent, authoritative answers regardless of where they're working or who's asking. For the docs team, it means the work we put into writing and maintaining docs.getdbt.com is doing more than it was before &mdash; reaching users where they actually are.

> <Lightbox src="/img/blog/2026-04-30-the-devil-is-in-the-docs/dwp-wear-chanel.png" width="55%"title="Emily salty users are faster using the dbt MCP server with the docs tools than without" />

Author: ```suggestion
<Lightbox src="/img/blog/2026-04-30-the-devil-is-in-the-docs/dwp-wear-chanel.png" width="55%"title="Emily loving that users are faster using the dbt MCP server with the docs tools than without" />
```
